### DISCUSSION TITLE
Laravel | Don't load relations were mentioned in protected $with = [];

### DISCUSSION THREAD

**Discussion by slisarenko** [laravel/framework#50964] [Q&A]
I have a query:
```
$users = User::query()->with([‘relation’])->get();
```

A *relation* described in User model:

```
public function relation(): MorphTo
{
    return $this->morphTo(__FUNCTION__, 'object_type', 'object_id');
}
```

Code: `$users->first()->relation` could return different models as it’s [Polymorphic Relationships][1], each model could have different [Eager Loading by Default][2].
```
class Book extends Model
{
    protected $with = ['authors', 'publications'];
}

class Magazine extends Model
{
    protected $with = ['shops'];
}
```
How to avoid loading these relations were defined in each model in `protected $with = [];`?

  [1]: https://laravel.com/docs/11.x/eloquent-relationships#polymorphic-relationships
  [2]: https://laravel.com/docs/11.x/eloquent-relationships#eager-loading-by-default

**Comment by slisarenko** [comment]
Personally for me, this looks similar to a case when we use the [morphWith](https://laravel-news.com/laravel-5-8-22) method, but in reverse. 
We need to avoid loading relations that are specified in:


https://github.com/laravel/framework/blob/81d8154874400f2625553a606c2d801d0f014a9a/src/Illuminate/Database/Eloquent/Model.php#L74-L79

Example:
```
class Post extends Model
{
    protected $with = ['user'];
}
```
```
$comments = Comment::query()
    ->with(['commentable' => function (MorphTo $morphTo) {
        $morphTo->morphWithout([Post::class => ['user']]);
    }])
    ->get();
```
In this case, we specify that if it's a Post model, we don't need to load the 'user' relation.

**Comment by ghost** [comment]
$keyType property (defualt 'int') on the model might prevent loading the relations (even if fetched from DB).
I had a situation when I forgot to put 'string' on that type and then the relation was null on the model.
